src/praisonai-agents/praisonaiagents/workflows/workflows.py

```
@@ -104,6 +104,29 @@ def _parse_json_output(output: Any, step_name: str = "step") -> Any:
     logger.debug(f"Could not parse JSON from step '{step_name}' output")
     return output
 
+def _substitute_action_variables(action: str, all_variables: dict, previous_output: Any, input: str) -> str:
```

Ruff (0.15.18)

[error] Function argument `input` is shadowing a Python builtin

(A002)